backend/app/services/im/control_service.py

```
@@ -0,0 +1,223 @@
+        db: Session | None,
+        *,
+        session: IMPrivateSession,
+        bot_purpose: str,
+        action_id: str,
+    ) -> dict[str, Any]:
+        action = await cache_manager.get(self._pending_action_key(action_id))
+        if (
+            not isinstance(action, dict)
+            or action.get("session_key") != session.session_key
+        ):
+            return self._error("确认操作已过期，请重新发起。", session, bot_purpose)
```

Ruff (0.15.18)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)